packages/legend-graph/src/helpers/DomainHelper.ts

```diff
@@ -49,3 +59,270 @@ export const getMilestoneTemporalStereotype = (
   });
   return stereotype;
 };
+
+export const milestoningPropertyGenerator = (
+  propertyOwner: PropertyOwner,
+  graph: PureModel,
+): void => {
+  propertyOwner.properties.forEach((property) => {
+    if (property.genericType.value.rawType instanceof Class) {
+      const milestonedStereotype = getMilestoneTemporalStereotype(
+        property.genericType.value.rawType,
+        graph,
+      );
+      switch (milestonedStereotype) {
+        case MILESTONING_STEROTYPES.BUSINESS_TEMPORAL: {
+          const dateProperty = new DerivedProperty(
+            `${property.name}`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONE),
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                graph.getClass(property.genericType.value.rawType.path),
+              ),
+            ),
+            property.owner,
+          );
+          dateProperty.parameters = [
```

Comment:
Instead of doing this could we build `VariableExpression` then serialize to JSON ? its a lot cleaner. lets do this for the rest as well.